lib/Doctrine/ORM/Tools/Console/ConsoleRunner.php

```diff
@@ -73,41 +77,9 @@ public static function createApplication($helperSetOrProvider, array $commands =
 
     public static function addCommands(Application $cli, ?EntityManagerProvider $entityManagerProvider = null): void
     {
-        if ($entityManagerProvider === null) {
-            $entityManagerProvider = new HelperSetManagerProvider($cli->getHelperSet());
-        }
-
-        $connectionProvider = new ConnectionFromManagerProvider($entityManagerProvider);
-
-        $cli->addCommands(
-            [
-                // DBAL Commands
-                new DBALConsole\Command\ImportCommand(),
-                new DBALConsole\Command\ReservedWordsCommand($connectionProvider),
-                new DBALConsole\Command\RunSqlCommand($connectionProvider),
-
-                // ORM Commands
-                new Command\ClearCache\CollectionRegionCommand($entityManagerProvider),
-                new Command\ClearCache\EntityRegionCommand($entityManagerProvider),
-                new Command\ClearCache\MetadataCommand($entityManagerProvider),
-                new Command\ClearCache\QueryCommand($entityManagerProvider),
-                new Command\ClearCache\QueryRegionCommand($entityManagerProvider),
-                new Command\ClearCache\ResultCommand($entityManagerProvider),
-                new Command\SchemaTool\CreateCommand($entityManagerProvider),
-                new Command\SchemaTool\UpdateCommand($entityManagerProvider),
-                new Command\SchemaTool\DropCommand($entityManagerProvider),
-                new Command\EnsureProductionSettingsCommand($entityManagerProvider),
-                new Command\ConvertDoctrine1SchemaCommand(),
-                new Command\GenerateRepositoriesCommand($entityManagerProvider),
-                new Command\GenerateEntitiesCommand($entityManagerProvider),
-                new Command\GenerateProxiesCommand($entityManagerProvider),
-                new Command\ConvertMappingCommand($entityManagerProvider),
-                new Command\RunDqlCommand($entityManagerProvider),
-                new Command\ValidateSchemaCommand($entityManagerProvider),
-                new Command\InfoCommand($entityManagerProvider),
-                new Command\MappingDescribeCommand($entityManagerProvider),
-            ]
-        );
+        $cli->addCommands(iterator_to_array(
```

Comment:
:thinking: why not call `addCommands` twice, conditionally the first time? Having a generator only for it to be immediately converted into an array feels a bit clumsy

Reply:
Sure.